homeassistant/components/blebox/binary_sensor.py

```diff
@@ -0,0 +1,38 @@
+"""BleBox sensor entities."""
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BleBoxEntity, create_blebox_entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a BleBox entry."""
+
+    create_blebox_entities(
+        hass,
+        config_entry,
+        async_add_entities,
+        BleBoxBinarySensorEntity,
+        "binary_sensors",
+    )
+
+
+class BleBoxBinarySensorEntity(BleBoxEntity, BinarySensorEntity):
+    """Representation of a BleBox sensor feature."""
+
+    @property
+    def is_on(self):
+        """Return the state."""
+        return self._feature.state
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return self._feature.device_class
```

Comment:
Normally we require components to have a static mapping between "component device class" and "home assistant device class"

For example:
```python
BLEEBOX_TO_HASS_DEVICE_CLASS : dict[str, BinarySensorDeviceClass] = {"moisture": BinarySensorDeviceClass.MOISTURE}
```

and then

```python
        return BLEEBOX_TO_HASS_DEVICE_CLASS[self._feature.device_class]
```

I see that this isn't applied in the existing code so maybe there is a reason. I am just pointing it out.

Reply:
Agree, I'll be using class based of `EntityDescription` to fulfill static mapping requirements. 
This way I'll be coherent with `sensor` platform.